### ISSUE TITLE
compilation error

### ISSUE THREAD

**Issue by aderzelle** [samtools/bcftools#1166]
Hello, trying to compile the latest version from github and running into this

```
vcffilter.c: In function ‘main_vcffilter’:
vcffilter.c:477:90: error: ‘VCF_OVERLAP’ undeclared (first use in this function); did you mean ‘VCF_OTHER’?
                         else if ( !strcasecmp(keys[i],"overlap") ) args->snp_gap_type |= VCF_OVERLAP;
                                                                                          ^~~~~~~~~~~
                                                                                          VCF_OTHER
vcffilter.c:477:90: note: each undeclared identifier is reported only once for each function it appears in
Makefile:136: recipe for target 'vcffilter.o' failed
make: *** [vcffilter.o] Error 1
```
thank you

**Comment by jkbonfield** [comment]
VCF_OVERLAP appeared with htslib 1.10.

This implies your compilation is picking up an earlier copy of htslib, possibly from a system directory.  

If that's the situation, it may be possible to forcibly get bcftools to build to another local copy with "./configure --with-htslib=DIR", but I believe it generally should default to local dirs if htslib is either a subdirectory of bcftools or a sibling directory.

**Comment by aderzelle** [comment]
Ok I removed my system htslib and system samtools, and the error disappeared but it still doesn't compile

```
/usr/bin/ld: cannot find -lperl
collect2: error: ld returned 1 exit status
Makefile:210: recipe for target 'bcftools' failed
make: *** [bcftools] Error 1

```
should I compile htslib before bcftools?

**Comment by pd3** [comment]
Please see detailed installation instructions here http://samtools.github.io/bcftools/howtos/install.html

**Comment by aderzelle** [comment]
As far as I can tell I am doing everything as per the doc

**Comment by aderzelle** [comment]
Ok, fixed by installing libperl, note it's not mentionned in requirements in the docs.

**Comment by jkbonfield** [comment]
If you're using ./configure before make, and configure isn't reporting a failure (it's not always that obvious in the wall of text sadly), then this is still a problem for us to fix.  I thought the perl support was optional and ideally configure should be either automatically disabling it with appropriate warnings, or reporting the problem and suggesting how to disable perl support (and then fail configure).

**Comment by aderzelle** [comment]
yes I did that, you could add libperl in the requirements list.

**Comment by arogozhnikov** [comment]
facing same issue on mac os. 
Configure does not complain. 
Disabling libperl filters does not help. 

Is there a recommended way I should get libperl on mac? Brew does not provide it

**Comment by pd3** [comment]
@arogozhnikov You say disabling did not help. Did you run `make clean && ./configure && make`? If this still result in an error, can you open a new issue and show the full outputs please.

**Comment by arogozhnikov** [comment]
@pd3 just tried, did not worked. Full output

```
make clean && ./configure && make
rm -f test/*.o test/*~ test/test-rbuf test/test-regidx
rm -f test/*.hex
rm -f plugins/*.so plugins/*.P
rm -rf plugins/*.dSYM
rm -f gmon.out *.o *.a *~ bcftools version.h
rm -rf *.dSYM test/*.dSYM
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for C compiler warning flags... -Wall
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking location of HTSlib source tree... ../htslib
checking shared library type... Darwin dylib
checking for library containing dlopen... none required
checking if the compiler accepts -rdynamic... yes
checking for zlib.h... yes
checking for inflate in -lz... yes
checking for library containing regcomp... none required
configure: creating ./config.status
config.status: creating config.mk
config.status: creating config.h
config.status: config.h is unchanged
echo '#define BCFTOOLS_VERSION "1.10"' > version.h
gcc -Wall -g -O2 -I. -I../htslib   -c -o main.o main.c
gcc -Wall -g -O2 -I. -I../htslib   -c -o vcfindex.o vcfindex.c
gcc -Wall -g -O2 -I. -I../htslib   -c -o tabix.o tabix.c
gcc -Wall -g -O2 -I. -I../htslib   -c -o vcfstats.o vcfstats.c
gcc -Wall -g -O2 -I. -I../htslib   -c -o vcfisec.o vcfisec.c
gcc -Wall -g -O2 -I. -I../htslib   -c -o vcfmerge.o vcfmerge.c
gcc -Wall -g -O2 -I. -I../htslib   -c -o vcfquery.o vcfquery.c
gcc -Wall -g -O2 -I. -I../htslib   -c -o vcffilter.o vcffilter.c
gcc -Wall -g -O2 -I. -I../htslib    -c -o filter.o filter.c
filter.c:2902:86: error: use of undeclared identifier 'VCF_OVERLAP'
            else if ( !strcasecmp(out[ival].key,"overlap") ) { out[ival].threshold = VCF_OVERLAP<<1; out[ival].is_str = 0; }
                                                                                     ^
1 error generated.
make: *** [filter.o] Error 1
```

- have you tried xcode?
No, I did not (and I did not need it anywhere else)

Perl itself is installed with brew. 

Don't see anything about arch in config.mk. Full paste:
```

prefix       = /usr/local
exec_prefix  = ${prefix}
bindir       = ${exec_prefix}/bin
libexecdir   = ${exec_prefix}/libexec
datarootdir  = ${prefix}/share
mandir       = ${datarootdir}/man

CC       = gcc
CPPFLAGS =
CFLAGS   =  -Wall -g -O2
LDFLAGS  =
LIBS     =

DYNAMIC_FLAGS = -rdynamic

USE_GPL =
GSL_LIBS =
PERL_CFLAGS =
PERL_LIBS =

PLATFORM   = Darwin
PLUGINS_ENABLED = yes
plugindir  = $(libexecdir)/bcftools
pluginpath = $(libexecdir)/bcftools
PLUGIN_EXT = .so

HTSDIR = ../htslib
include $(HTSDIR)/htslib.mk
include $(HTSDIR)/htslib_static.mk
HTSLIB = $(HTSDIR)/libhts.a
HTSLIB_LIB = $(HTSLIB) $(HTSLIB_static_LIBS)
HTSLIB_DLL = $(HTSDIR)/
HTSLIB_LDFLAGS = $(HTSLIB_static_LDFLAGS)
W32_PLUGIN_LIBS = libbcftools.a $(HTSLIB_DLL) $(ALL_LIBS)
BGZIP = $(HTSDIR)/bgzip
TABIX = $(HTSDIR)/tabix
HTSLIB_CPPFLAGS = -I../htslib
#HTSLIB_LDFLAGS = -L../htslib
#HTSLIB_LIB = -lhts
#W32_PLUGIN_LIBS = libbcftools.a $(HTSLIB_LDFLAGS) $(HTSLIB_LIB) $(ALL_LIBS)
```

Probably that's all I can help with

**Comment by pd3** [comment]
Apparently you did not run the commands as suggested in the install page given above. The VCF_OVERLAP complaint comes from trying to compile with an older version of htslib.
````
git clone git://github.com/samtools/htslib.git
git clone git://github.com/samtools/bcftools.git
cd bcftools
autoheader && autoconf && ./configure
make
````

**Comment by arogozhnikov** [comment]
@pd3 you're correct, I got some older htslib. Now it builds successfully, thanks!
